### DISCUSSION TITLE
Renovate with local paths dependency

### DISCUSSION THREAD

**Discussion by UdiBen** [renovatebot/renovate#26427] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

I have 2 projects - `send-email` and `common`.
send-email depends on common, and is used as a local file:
```
    "dependencies": {
        "common": "file:../common/build"
   }
```
And I have a preinstall script in send-email that is making sure that common gets built before the installation:
```
"scripts": {
    ...    
   "preinstall": "cd ../common && yarn install && yarn build"
},
```

The problem is that when renovate is creating a PR - `yarn install` is failing in send-email:
`error Package "common" refers to a non-existing file '"/tmp/renovate/repos/github/.../application/lambdas/common/build"'.`

Am I missing anything? Is renovate expected to honor lifecycle scripts? 
 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
The hosted app does not support install scripts due to security controls

**Reply by UdiBen** [reply]
Thanks @rarkins
Is there any workaround that will allow building common prior to installing the other project?

**Reply by rarkins** [reply]
Not to my knowledge. In short we can't allow arbitrary scripts to execute

**Comment by kimsey0** [comment]
We had a similar problem and fixed it by adding an [exports](https://nodejs.org/api/packages.html#exports) field to the `package.json` file in the dependency (in your case `common/package.json`):

```
"exports": {
  ".": {
    "import": "./build/index.js",
    "types": "./build/index.d.ts", // If using TypeScript.
    "default": "./build/index.js"
  },
  "./*": {
    "import": "./build/*.js",
    "types": "./build/*.d.ts", // If using TypeScript.
    "default": "./build/*.js"
  },
  // Additional objects if you have more index files.
}
```

and then pointing our file dependency to the root folder instead of the `build` subfolder:

```
"dependencies": {
  "common": "file:../common"
}
```

(Since we're using TypeScript, we also had to change to `"moduleResolution": "Bundler"` in our `tsconfig.json`. Your mileage may vary.)
